homeassistant/components/squeezebox/media_player.py

```diff
@@ -182,133 +180,52 @@ async def async_service_handler(service):
     return True
 
 
-class LogitechMediaServer:
-    """Representation of a Logitech media server."""
-
-    def __init__(self, hass, host, port, username, password):
-        """Initialize the Logitech device."""
-        self.hass = hass
-        self.host = host
-        self.port = port
-        self._username = username
-        self._password = password
-
-    async def create_players(self):
-        """Create a list of devices connected to LMS."""
-        result = []
-        data = await self.async_query("players", "status")
-        if data is False:
-            return None
-        for players in data.get("players_loop", []):
-            player = SqueezeBoxDevice(self, players["playerid"], players["name"])
-            await player.async_update()
-            result.append(player)
-        return result
-
-    async def async_query(self, *command, player=""):
-        """Abstract out the JSON-RPC connection."""
-        auth = (
-            None
-            if self._username is None
-            else aiohttp.BasicAuth(self._username, self._password)
-        )
-        url = f"http://{self.host}:{self.port}/jsonrpc.js"
-        data = json.dumps(
-            {"id": "1", "method": "slim.request", "params": [player, command]}
-        )
-
-        _LOGGER.debug("URL: %s Data: %s", url, data)
-
-        try:
-            websession = async_get_clientsession(self.hass)
-            with async_timeout.timeout(TIMEOUT):
-                response = await websession.post(url, data=data, auth=auth)
-
-                if response.status != HTTP_OK:
-                    _LOGGER.error(
-                        "Query failed, response code: %s Full message: %s",
-                        response.status,
-                        response,
-                    )
-                    return False
-
-                data = await response.json()
-
-        except (asyncio.TimeoutError, aiohttp.ClientError) as error:
-            _LOGGER.error("Failed communicating with LMS: %s", type(error))
-            return False
-
-        try:
-            return data["result"]
-        except AttributeError:
-            _LOGGER.error("Received invalid response: %s", data)
-            return False
-
-
 class SqueezeBoxDevice(MediaPlayerEntity):
-    """Representation of a SqueezeBox device."""
+    """
+    Representation of a SqueezeBox device.
 
-    def __init__(self, lms, player_id, name):
+    Wraps a pysqueezebox.Player() object.
+    """
+
+    def __init__(self, hass, player):
```

Comment:
Don't pass in `hass`. It will be set on the entity when the entity has been added to home assistant. This means you can access `hass` with `self.hass`, after the entity has been added to home assistant.

Reply:
```suggestion
    def __init__(self, player):
```